pkg/volume/util/operationexecutor/operation_executor.go

```diff
@@ -330,6 +335,13 @@ type operationExecutor struct {
 	// pendingOperations keeps track of pending attach and detach operations so
 	// multiple operations are not started on the same volume
 	pendingOperations nestedpendingoperations.NestedPendingOperations
+
+	// recorder is used to record events in the API server
+	recorder record.EventRecorder
+
+	// represents a volume that should be attached to this node and
+	// mounted to the PodName.
+	volumeToMount VolumeToMount
```

Comment:
Good point @saad-ali.  This needs to be a pointer.

So, my next question is, we really ought to be eventing two subjects in most cases:

1.  The pod that mounts the volume
2.  The volume itself

For detach, we can only event the volume -- are there any other cases that we should event the volume for?

Another Q: should we event the pod for attach?